core/src/builtin/value.rs

```diff
@@ -0,0 +1,71 @@
+use crate::AllocId;
+use crate::ContextRef;
+use crate::OpRef;
+use crate::Type;
+
+pub struct Value<T: Into<Type> + From<Type>> {
+    op_id: AllocId,
+    ty: T,
+}
+
+impl<T: Into<Type> + From<Type> + Copy> Value<T> {
+    fn get_context(&self) -> ContextRef {
```

Comment:
Also you're missing pub on functions, and that's probably why the CI is complaining about dead code.

Reply:
Done, also added some tests